src/com/company/M2/Task45.java

```diff
@@ -0,0 +1,29 @@
+package com.company.M2;
+
+public class Task45 {
+    public static void main(String[] args) {
+        System.out.println("[null]: " + makeHalfOfName(null));
+        System.out.println("[empty name]: " + makeHalfOfName(""));
+        System.out.println("Magicoraxi: " + makeHalfOfName("Magicoraxi"));
+        System.out.println("Javametro: " + makeHalfOfName("Javametro"));
+    }
+
+    public static String makeHalfOfName(String name) {
+        if (name == null || name.isBlank()) {
+            return "";
+        }
+        int size;
+        if (name.length() % 2 == 0) {
+            size = name.length() / 2;
+        } else {
+            size = (name.length() - 1) / 2;
+        }
+        char[] chars = new char[size];
+        int poz = 0;
+        for (int i = 0; i < size; i++) {
+            chars[poz] = name.charAt(i);
+            poz++;
+        }
+        return String.valueOf(chars);
```

Comment:
Let's leave this solution as varian1, and implent one more variant with using appropriate methods for getting substring  from the String class methods.

Reply:
fixed